src/app/externs/backendapi.js

```diff
@@ -218,3 +218,17 @@ backendApi.AppNameValiditySpec;
  * }}
  */
 backendApi.AppNameValidity;
+
+/**
+ * @typedef {{
+ *    protocols: !Array<string>
+ * }}
+ */
+backendApi.Protocols;
+
+/**
+ * @typedef {{
+ *    isExternal: boolean
+ * }}
+ */
+backendApi.ProtocolsSpec;
```

Comment:
To meet half way I've added ProtocolsSpec to frontend so it can be used for documentation and in backend I've added comment explaining `isExternal`.